grype/pkg/syft_sbom_provider.go

```diff
@@ -23,67 +34,120 @@ func (e errEmptySBOM) Error() string {
 }
 
 func syftSBOMProvider(userInput string, config ProviderConfig) ([]Package, Context, error) {
-	reader, err := getSBOMReader(userInput)
+	s, err := getSBOM(userInput, config)
```

Comment:
Nice! I like that this is simpler without the extra step. A good candidate we can look at to what we might want to see eventually exported from syft since we had to write our own function here.